src/EventStore.Core.XUnit.Tests/Services/Archive/Storage/ArchiveStorageReaderTests.cs

```diff
@@ -1,16 +1,86 @@
 // Copyright (c) Event Store Ltd and/or licensed to Event Store Ltd under one or more agreements.
 // Event Store Ltd licenses this file to you under the Event Store License v2 (see LICENSE.md).
 
+using System;
 using System.IO;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using EventStore.Core.Services.Archive;
+using EventStore.Core.Services.Archive.Storage.Exceptions;
+using FluentStorage.Utils.Extensions;
 using Xunit;
 
 namespace EventStore.Core.XUnit.Tests.Services.Archive.Storage;
 
+[Collection("ArchiveStorageTests")]
 public class ArchiveStorageReaderTests : ArchiveStorageTestsBase<ArchiveStorageReaderTests> {
+	[Theory]
+	[InlineData(StorageType.FileSystem)]
+	[InlineData(StorageType.S3, Skip = SkipS3)]
+	public async Task can_read_chunk_entirely(StorageType storageType) {
+		var sut = CreateReaderSut(storageType);
+
+		// create a chunk and upload it
+		var chunkPath = CreateLocalChunk(0, 0);
+		await CreateWriterSut(storageType).StoreChunk(chunkPath, CancellationToken.None);
+
+		// read the local chunk
+		var localContent = await File.ReadAllBytesAsync(chunkPath);
+
+		// read the uploaded chunk
+		using var chunkStream = await sut.GetChunk(chunkPath, CancellationToken.None);
+		var chunkStreamContent = chunkStream.ToByteArray();
+
+		// then
+		Assert.Equal(localContent, chunkStreamContent);
```

Comment:
in the past, i've measured the time to compare byte arrays using `Assert.Equal` and it was much faster to use Span.SequenceEqual: `Assert.True(localContent.Span.SequenceEqual(chunkStreamContent))`

i'm not sure if that's still the case